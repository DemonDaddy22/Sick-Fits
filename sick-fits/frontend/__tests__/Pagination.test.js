import { mount } from 'enzyme';
import Router from 'next/router';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import Pagination, { PAGINATION_QUERY } from '../components/Pagination';
import { MockedProvider } from 'react-apollo/test-utils';

Router.router = {
    push: () => { },
    prefetch: () => { },
};

const makeMocksFor = length => [
    {
        request: { query: PAGINATION_QUERY },
        result: {
            data: {
                itemsConnection: {
                    __typename: 'aggregate',
                    aggregate: {
                        __typename: 'count',
                        count: length
                    }
                }
            }
        }
    }
];

describe('<Pagination />', () => {

    it('renders loading', async () => {
        const wrapper = mount(<MockedProvider mocks={makeMocksFor(1)}>
            <Pagination page={1} />
        </MockedProvider>);
        // console.log(wrapper.debug());
        expect(wrapper.text()).toContain('Loading');
    });

    it('renders pagination for 18 items', async () => {
        const wrapper = mount(<MockedProvider mocks={makeMocksFor(18)}>
            <Pagination page={1} />
        </MockedProvider>);

        await wait();
        wrapper.update();
        const pagination = wrapper.find('[data-test="pagination"]');
        expect(toJSON(pagination)).toMatchSnapshot();
        expect(wrapper.find('.totalPages').text()).toEqual('5');
    });

    it('disables prev button on first page', async () => {
        const wrapper = mount(<MockedProvider mocks={makeMocksFor(18)}>
            <Pagination page={1} />
        </MockedProvider>);

        await wait();
        wrapper.update();
        expect(wrapper.find('a.prev').prop('aria-disabled')).toEqual(true);
    });

    it('disables next button on last page', async () => {
        const wrapper = mount(<MockedProvider mocks={makeMocksFor(18)}>
            <Pagination page={5} />
        </MockedProvider>);

        await wait();
        wrapper.update();
        expect(wrapper.find('a.next').prop('aria-disabled')).toEqual(true);
    });

    it('enables all buttons on middle page', async () => {
        const wrapper = mount(<MockedProvider mocks={makeMocksFor(18)}>
            <Pagination page={3} />
        </MockedProvider>);

        await wait();
        wrapper.update();
        expect(wrapper.find('a.prev').prop('aria-disabled')).toEqual(false);
        expect(wrapper.find('a.prev').prop('aria-disabled')).toEqual(false);
    });
});