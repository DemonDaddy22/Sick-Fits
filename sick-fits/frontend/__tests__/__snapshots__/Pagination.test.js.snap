// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> renders pagination for 18 items 1`] = `
Array [
  <PaginationStyles
    data-test="pagination"
  >
    <div
      className="PaginationStyles-aduuar-0 ewJsCc"
      data-test="pagination"
    >
      <SideEffect(Head)>
        <Head />
      </SideEffect(Head)>
      <Link
        href={
          Object {
            "pathname": "shop",
            "query": Object {
              "page": 0,
            },
          }
        }
        prefetch={true}
      >
        <a
          aria-disabled={true}
          className="prev"
          href="shop?page=0"
          onClick={[Function]}
        >
          &lt;
           Prev
        </a>
      </Link>
      <p>
        Page 
        1
         of 
        <span
          className="totalPages"
        >
          5
        </span>
      </p>
      <Link
        href={
          Object {
            "pathname": "shop",
            "query": Object {
              "page": 2,
            },
          }
        }
        prefetch={true}
      >
        <a
          aria-disabled={false}
          className="next"
          href="shop?page=2"
          onClick={[Function]}
        >
          Next 
          &gt;
        </a>
      </Link>
    </div>
  </PaginationStyles>,
  <div
    className="PaginationStyles-aduuar-0 ewJsCc"
    data-test="pagination"
  >
    <SideEffect(Head)>
      <Head />
    </SideEffect(Head)>
    <Link
      href={
        Object {
          "pathname": "shop",
          "query": Object {
            "page": 0,
          },
        }
      }
      prefetch={true}
    >
      <a
        aria-disabled={true}
        className="prev"
        href="shop?page=0"
        onClick={[Function]}
      >
        &lt;
         Prev
      </a>
    </Link>
    <p>
      Page 
      1
       of 
      <span
        className="totalPages"
      >
        5
      </span>
    </p>
    <Link
      href={
        Object {
          "pathname": "shop",
          "query": Object {
            "page": 2,
          },
        }
      }
      prefetch={true}
    >
      <a
        aria-disabled={false}
        className="next"
        href="shop?page=2"
        onClick={[Function]}
      >
        Next 
        &gt;
      </a>
    </Link>
  </div>,
]
`;
